### ISSUE TITLE
onTileLoad Event

### ISSUE THREAD

**Issue by octagonalsquare** [react-native-maps/react-native-maps#3693]
# Feature request

A onTileLoad or similar event that passes the x,y,z tile format to a function.

## Why it is needed

The tile-based system obviously exists in the map as you have the urlTemplate/pathTemplate for the UrlTile and LocalTile respectively. It would allow us to get more than just tiles that are based off of the x,y,z system. I have an api that returns GeoJson based on the tiles. I can calculate the x,y,z for the tiles I need, but calculating them manually, doing multiple async calls, and then putting them in state and processing them for display is far slower and more complicated.

## Possible implementation

<MapView
    onTileLoad={ ( tile ) => getTiledData( {x,y,z} ) } />

**Comment by IPWright83** [comment]
@octagonalsquare sorry, bit of a tangent - but have you managed to find a way to access those parameters? I want to log `x,y,z` for some debugging but can't work out how to access them.

**Comment by octagonalsquare** [comment]
> @octagonalsquare sorry, bit of a tangent - but have you managed to find a way to access those parameters? I want to log `x,y,z` for some debugging but can't work out how to access them.

unfortunately no. If I had I would be using them. But you can calculate them, though probably not exactly what the UrlTile uses. The functions are as follows:

`calcZoom = (region) => {
	let z = Math.round(Math.log(360 / region.longitudeDelta) / Math.LN2);
	return z;
}
lon2tile = (lon,zoom) => { 
	return (Math.floor((lon+180)/360*Math.pow(2,zoom))); 
}
lat2tile = (lat,zoom) => {
		return (Math.floor((1-Math.log(Math.tan(lat*Math.PI/180) + 1/Math.cos(lat*Math.PI/180))/Math.PI)/2 *Math.pow(2,zoom)));
}`

Use calcZoom to get your z, then the others to get x and y

The problem comes from needing multiple tiles to display. You then need to calculate all the tiles you need to show on screen. Once you do that, then you have to do fetch calls for all of them separately or create an array and call them with Promise.all which gets clunky and hard to follow very quickly
